### PR TITLE
added extra docs on metrics for cpu/memory for pod/container in the kubernetes module

### PR DESCRIPTION
**Pull request by gsantoro**
## What does this PR do?

Add some extra docs to explain why some metrics might be missing.

## Why is it important?

Some clients have complained that some metrics were missing and didn't understand why

## Checklist

- [x] I have made corresponding changes to the documentation
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates https://github.com/elastic/beats/pull/32539

### CODE REVIEW

File: metricbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -43262,7 +43262,7 @@ format: percent
 *`kubernetes.container.cpu.usage.limit.pct`*::
 +
 --
-CPU usage as a percentage of the defined limit for the container (or total node allocatable CPU if unlimited)
+CPU usage as a percentage of the defined limit for the container (or total node allocatable CPU if unlimited). If the container CPU limits are missing and the `node` and `state_node` metricsets are both disabled or not configured on that node, this metric will be missing entirely.
```
Comment (tetianakravchenko) [inline metricbeat/docs/fields.asciidoc:43265]: what do you mean by `or not configured`?

File: metricbeat/docs/fields.asciidoc [hunk 2]
```diff
@@ -43390,7 +43390,7 @@ format: percent
 *`kubernetes.container.memory.usage.limit.pct`*::
 +
 --
-Memory usage as a percentage of the defined limit for the container (or total node allocatable memory if unlimited)
+Memory usage as a percentage of the defined limit for the container (or total node allocatable memory if unlimited). If the container Memory limits are missing and the `node` and `state_node` metricsets are both disabled or not configured on that node, this metric might will be missing entirely.
```
Comment (tetianakravchenko) [inline metricbeat/docs/fields.asciidoc:43393]:
```suggestion
Memory usage as a percentage of the defined limit for the container (or total node allocatable memory if unlimited). If the container Memory limits are missing and the `node` and `state_node` metricsets are both disabled or not configured on that node, this metric will be missing entirely.
```

File: metricbeat/docs/fields.asciidoc (hunk 1 above)
Comment (gsantoro) [reply metricbeat/docs/fields.asciidoc:43265]: `not configured` = `not in the configuration`. any better idea how to convey that?

**Comment by gsantoro** [comment]
> @gsantoro are you going to backport this pr to older versions?

we are already too late for the FF for 7.17.6 and 8.4 so I am not sure we should open any backport PR.

File: metricbeat/docs/fields.asciidoc (hunk 1 above)
Comment (tetianakravchenko) [reply metricbeat/docs/fields.asciidoc:43265]: in this case ```the `node` and `state_node` metricsets are both disabled``` = `not configured` ? maybe then just leave first part?

File: metricbeat/docs/fields.asciidoc [hunk 3]
```diff
@@ -43390,7 +43390,7 @@ format: percent
 *`kubernetes.container.memory.usage.limit.pct`*::
 +
 --
-Memory usage as a percentage of the defined limit for the container (or total node allocatable memory if unlimited)
+Memory usage as a percentage of the defined limit for the container (or total node allocatable memory if unlimited). If the container Memory limits are missing and the `node` and `state_node` metricsets are both disabled or not configured on that node, this metric will be missing entirely.
```
Comment (tetianakravchenko) [inline metricbeat/docs/fields.asciidoc:43393]: the same here regarding `or not configured`, and for pod metrics.

**Review by tetianakravchenko — APPROVED** [review]

**Pull request merged**